### PR TITLE
Create Pcli

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**

This creates a prysm specific cli, through which we can perform all the useful eth2 utilities.
Analogous to zcli,lcli,ncli for the other eth2 clients. This utility will help in debugging any state 
root bugs. Currently it just performs manual state transitions. 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

N.A

### CODE REVIEW

File: tools/pcli/main.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			log.Infof("Performing State Transition with a block root of %#x and pre state root of %#x",
+				blkRoot, preStateRoot)
+			postState, err := state.ExecuteStateTransition(context.Background(), stateObj, block)
+			if err != nil {
+				log.Fatal(err)
+			}
+			postRoot, err := postState.HashTreeRoot(context.Background())
+			log.Infof("Finished Performing State Transition with Post state root of %#x", postRoot)
+
+			// diff the state if a post state is provided
```
Comment (farazdagi) [inline tools/pcli/main.go:114]:
```suggestion
			// Diff the state if a post state is provided.
```

File: tools/pcli/main.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		},
+	},
+	}
+	if err := app.Run(os.Args); err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+// fetch data and unmarshal to provided data structure from file
```
Comment (farazdagi) [inline tools/pcli/main.go:135]:
```suggestion
// dataFetcher fetches and unmarshals data from file to provided data structure.
```

File: beacon-chain/core/state/BUILD.bazel [hunk 3]
```diff
@@ -12,11 +12,12 @@ go_library(
     importpath = "github.com/prysmaticlabs/prysm/beacon-chain/core/state",
     visibility = [
         "//beacon-chain:__subpackages__",
+        "//endtoend:__pkg__",
         "//shared/interop:__pkg__",
         "//shared/testutil:__pkg__",
         "//tools/benchmark-files-gen:__pkg__",
         "//tools/genesis-state-gen:__pkg__",
-        "//endtoend:__pkg__",
```
Comment (shayzluf) [inline beacon-chain/core/state/BUILD.bazel:19]: is there a reason to replace end to end?

Comment (nisdas) [reply beacon-chain/core/state/BUILD.bazel:19]: not replaced its just shifted above ^

File: tools/pcli/main.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	app.Name = "pcli"
+	app.Usage = "A command line utility to run eth2 specific commands"
+	app.Version = version.GetVersion()
+	app.Commands = []*cli.Command{{
+		Name:     "state-transition",
+		Category: "state-transition",
+		Usage:    "Subcommand to run manual state transitions",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "blockPath",
```
Comment (shayzluf) [inline tools/pcli/main.go:43]:
```suggestion
				Name:        "block-path",
```

File: tools/pcli/main.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+				Usage:       "Path to block file(ssz)",
+				Destination: &blockPath,
+			},
+			&cli.StringFlag{
+				Name:        "preStatePath",
```
Comment (shayzluf) [inline tools/pcli/main.go:48]: our cli flags are using dashes usually
```suggestion
				Name:        "pre-state-path",
```

File: tools/pcli/main.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+				Usage:       "Path to pre state file(ssz)",
+				Destination: &preStatePath,
+			},
+			&cli.StringFlag{
+				Name:        "expectedPostStatePath",
```
Comment (shayzluf) [inline tools/pcli/main.go:53]:
```suggestion
				Name:        "expected-post-state-path",
```

File: tools/pcli/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+*State Transition Subcommand:*
+   pcli state-transition - Subcommand to run manual state transitions
+
+*State Transition Usage:*:
+   pcli state-transition [command options] [arguments...]
+
+
+*State Transition Flags:*
+   --blockPath value              Path to block file(ssz)
```
Comment (shayzluf) [inline tools/pcli/README.md:31]:
```suggestion
   --block-path value              Path to block file(ssz)
```

File: tools/pcli/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+   --preStatePath value           Path to pre state file(ssz)
```
Comment (shayzluf) [inline tools/pcli/README.md:32]:
```suggestion
   --pre-state-path value           Path to pre state file(ssz)
```

File: tools/pcli/README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+   --expectedPostStatePath value  Path to expected post state file(ssz)
```
Comment (shayzluf) [inline tools/pcli/README.md:33]:
```suggestion
   --expected-post-state-path value  Path to expected post state file(ssz)
```

File: beacon-chain/core/state/BUILD.bazel (hunk 3 above)
Comment (shayzluf) [reply beacon-chain/core/state/BUILD.bazel:19]: missed it. sorry

File: tools/pcli/main.go [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+			}
+			stateObj, err := stateTrie.InitializeFromProto(preState)
+			if err != nil {
+				log.Fatal(err)
+			}
+			preStateRoot, err := stateObj.HashTreeRoot(context.Background())
+			if err != nil {
+				log.Fatal(err)
+			}
+			log.Infof("Performing State Transition with a block root of %#x and pre state root of %#x",
```
Comment (terencechain) [inline tools/pcli/main.go:105]:
```suggestion
			log.Infof("Performing state transition with a block root of %#x and pre state root of %#x",
```

File: tools/pcli/main.go [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+				blkRoot, preStateRoot)
+			postState, err := state.ExecuteStateTransition(context.Background(), stateObj, block)
+			if err != nil {
+				log.Fatal(err)
+			}
+			postRoot, err := postState.HashTreeRoot(context.Background())
+			log.Infof("Finished Performing State Transition with Post state root of %#x", postRoot)
```
Comment (terencechain) [inline tools/pcli/main.go:112]:
```suggestion
			log.Infof("Finished state transition with post state root of %#x", postRoot)
```

File: tools/pcli/main.go (hunk 10 above)
Comment (terencechain) [reply tools/pcli/main.go:105]: On the second thought, can we use `WithFields` and also log the pre state slot number and the block slot number?

**Pull request merged**